LeanMapperQuery/Query.php

```diff
@@ -506,11 +515,29 @@ public function applyQuery(Fluent $fluent, IMapper $mapper)
 			list($method, $args) = $call;
 			call_user_func_array([$this, $method], $args);
 		}
+
+		if ($this->limit !== NULL) {
+			$fluent->limit($this->limit);
+		}
+
+		if ($this->offset !== NULL) {
+			$fluent->offset($this->offset);
+		}
+
 		// Reset fluent.
 		$this->fluent = NULL;
+		$this->primaryTable = NULL;
 		return $fluent;
 	}
 
+	/**
+	 * @return bool
+	 */
+	public function hasLimitOrOffset()
```

Comment:
Tohle bych přejmenoval na `junctionQueryNeeded`. To, že se to týká `LIMITu` a `OFFSETu` je vlastně implementační detail.